src/main/java/org/apache/commons/imaging/formats/tiff/datareaders/DataReaderStrips.java

```diff
@@ -369,8 +371,15 @@ public TiffRasterData readRasterData(final Rectangle subImage)
             rasterWidth = width;
             rasterHeight = height;
         }
-        final float[] rasterData = new float[rasterWidth * rasterHeight];
 
+        float[] rasterDataFloat = null; // new float[rasterWidth * rasterHeight];
+        int[] rasterDataInt = null;
+
+        if (sampleFormat == TiffTagConstants.SAMPLE_FORMAT_VALUE_IEEE_FLOATING_POINT) {
+            rasterDataFloat = new float[rasterWidth * rasterHeight];
+        } else {
+            rasterDataInt = new int[rasterWidth * rasterHeight];
+        }
```

Comment:
@gwlucastrig would it make sense to have a FloatDataReaderStrips, and an IntegerDataReaderStrips? Then just choose which one to use, and remove these `if/else`'s in the code?

Same for DataReaderTiled, maybe a FloatDataReaderTiled, and an IntegerDataReaderTiled?

Or pass an object that is responsible for calculating the values for float/int... just so we don't have these `if/else`...

Reply:
This would probably eliminate the need for the enum above. Although I haven't tried a different design in the code/IDE yet...

Reply:
open to suggestions.  when I first wrote this, I had two very similar methods named readRasterDataInt() and readRasterDataFloat() with a branch inside readRasterData() based on data type.   The individual methods were a bit redundant, but the code was cleaner.  I will return the code to that approach.

In a future pull request, I intend to go over the various class dependencies and relationships.  They are a bit convoluted, and I am reluctant to change too much at once. If we still think it's appropriate, I can try splitting DataReaderStrips and Tiles into type-specific variations.